### PR TITLE
Build: Fix the local storybook

### PR DESCRIPTION
**Pull request by ndelangen**
I found the local dev of storybook was broken

### CODE REVIEW

File: code/lib/builder-vite/src/list-stories.ts [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
+      const pattern = path.join(directory, files);
+      const absolutePattern = path.isAbsolute(pattern)
+        ? pattern
+        : path.join(options.configDir, pattern);
 
-        return glob(slash(absolutePattern), { follow: true });
-      })
-    )
-  ).reduce((carry, stories) => carry.concat(stories.map(normalizePath)), []);
+      return glob.sync(slash(absolutePattern), { follow: true });
```
Comment (ndelangen) [inline code/lib/builder-vite/src/list-stories.ts:22]: So I discovered (by console.logging) that the output of `glob` (in contrast to what the type definitions say) is actually returning a Class instance.

I can't understand why.
Using the sync method does work.

It'd be preferred if we could stay on an async API.

**Comment by ndelangen** [comment]
Seems what I'm trying to fix here, actually does happen on `next`:
https://app.circleci.com/pipelines/github/storybookjs/storybook/50258/workflows/7e868010-e659-4021-80b7-ecfdad43975b/jobs/531998

File: code/lib/builder-vite/src/list-stories.ts (hunk 1 above)
Comment (ndelangen) [reply code/lib/builder-vite/src/list-stories.ts:22]: I've tried a few alternate strategies as well:

```
const glob = require('glob');

const glob = await import('glob');

import * as glob from 'glob';
```

None of that made any difference, the only way I could find to get it to return an array was with the sync method.

Comment (ndelangen) [reply code/lib/builder-vite/src/list-stories.ts:22]: I could also try to prebundle the `glob` package, I haven't tried that yet.

Comment (tmeasday) [reply code/lib/builder-vite/src/list-stories.ts:22]: @ndelangen is `glob.sync` sync? (I would guess so). If so, should we drop the `Promise.all`?

It's not ideal not to do this in parallel though.

Comment (ndelangen) [reply code/lib/builder-vite/src/list-stories.ts:22]: Yes it is sync, indeed.

Comment (JReinhold) [reply code/lib/builder-vite/src/list-stories.ts:22]: What if we actually used the Glob class instead (and import it as intended)? Based on the docs, it looks like `new Glob(..).walk()` is the same as `glob(...)`, and is async.

https://github.com/isaacs/node-glob#gwalk

**Review by JReinhold — APPROVED** [review]

**Comment by ndelangen** [comment]
Final findings:

glob is being resolved to the wrong location in the monorepo workspace, for some reason.
I logged `require.resolve('glob/package.json')` and the output was wild! it resolved glob v7 from `framework/react-vite` 🤯 

This is just really really wrong, and I don't know how to properly fix it.
I suspect it's something symlink/node related that we do for linked sandboxes.

I made `glob` be bundled in, that way it can't resolve the wrong thing.

@tmeasday I'd love to hear your ideas on what might be causing the issue and how to properly fix, you're our symlinking expert ;)

For now, I'm going to unblock next by merging this 🎉

**Pull request merged**